### DISCUSSION TITLE
Dagrun reports in markdown

### DISCUSSION THREAD

**Discussion by xeor** [apache/airflow#28035] [General]
I want to generate a page, or a doc, or just a table with some key=values that are available when clicking on the dagrun. Is there anyway to do this?

I am thinking about generating a markdown file and put it somewhere that `doc_md` can pick it up, per dag_run. I'm not sure if this functionality is already built in tho, I can't find it.

The `doc_md` seams to go trough a jinja2 renderer, but I'm not sure what context is available. It doesn't seam that any of the normal template variables are available.

What would be perfect is a table, with just some hard facts that is gathered when the run ends, or maybe even when it is running. Example would be a "fact" about "scanned files: 123", "link to workdir: /..../...".

Possible?

update:
I poked around a little bit, and it looks like you can use javascript in `doc_md`. The dag-run-id is available in some views as `$('meta[name="dag_run_id"]')`, so it might be possible to create some javascript that collects the required data.. This feels very hackish tho :(

**Comment by potiuk** [comment]
The `doc_md` like all other params are evaluated during parsing . It gets rendered via jinja without any context (so only built-in jinja features such as looping etc. are available. You can basically program your doc this way - but it's not "dynamic" in the sense that it wil look differently for different dag_run.

If you want to do some exta procesing after dag or task finishes, you can use new listeners in 2.5.0 but you cannot report them as "facts" without writing custom plugin, https://airflow.apache.org/docs/apache-airflow/stable/listeners.html.

If you would like to elaborate and add such feature - feel free. PRs are most welcome, or you do not feel like you want to join > 2200 contributors to Airflow, you can create a feature issue with more comprehensive and complete information. If you get interest of others, then maybe someone will feel like it and implement it.

**Reply by sou-joshi** [reply]
@xeor looking for the same. Did you find any other ways of implementing? Or raised a feature request?

**Reply by xeor** [reply]
@sou-joshi there are some neat features coming in the next release, including markdown in notes (https://github.com/apache/airflow/issues/28167). Check out https://github.com/apache/airflow/discussions/28448 for how to change the note using the api as well. These two together are solving my usecase :)
